src/datasources/push-notifications-api/firebase-cloud-messaging-api.service.ts

```diff
@@ -0,0 +1,141 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { CacheRouter } from '@/datasources/cache/cache.router';
+import {
+  CacheService,
+  ICacheService,
+} from '@/datasources/cache/cache.service.interface';
+import { HttpErrorFactory } from '@/datasources/errors/http-error-factory';
+import {
+  NetworkService,
+  INetworkService,
+} from '@/datasources/network/network.service.interface';
+import { IPushNotificationsApi } from '@/domain/interfaces/push-notifications-api.interface';
+import { Inject, Injectable } from '@nestjs/common';
+import { FirebaseNotification } from '@/datasources/push-notifications-api/entities/firebase-notification.entity';
+
+// TODO: Refactor to use JwtService
+import jwt from 'jsonwebtoken';
+
+@Injectable()
+export class FirebaseCloudMessagingApiService implements IPushNotificationsApi {
+  private static readonly OAuth2TokenUrl =
+    'https://oauth2.googleapis.com/token';
+  private static readonly Scope =
+    'https://www.googleapis.com/auth/firebase.messaging';
+
+  private readonly baseUrl: string;
+  private readonly project: string;
+  private readonly clientEmail: string;
+  private readonly privateKey: string;
+
+  constructor(
+    @Inject(NetworkService)
+    private readonly networkService: INetworkService,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(CacheService)
+    private readonly cacheService: ICacheService,
+    private readonly httpErrorFactory: HttpErrorFactory,
+  ) {
+    this.baseUrl = this.configurationService.getOrThrow<string>(
+      'pushNotifications.baseUri',
+    );
+    this.project = this.configurationService.getOrThrow<string>(
+      'pushNotifications.project',
+    );
+    // Service account credentials are used for OAuth2 assertion
+    this.clientEmail = this.configurationService.getOrThrow<string>(
+      'pushNotifications.serviceAccount.clientEmail',
+    );
+    this.privateKey = this.configurationService.getOrThrow<string>(
+      'pushNotifications.serviceAccount.privateKey',
+    );
```

Comment:
I guess these environment variables will be mandatory, unless we use a flag to signal the push notifications availability. I'm fine either way, but if it won't be behind a flag we need to add these variables to `.env.sample`/`configuration.validator.spec.ts`/`safe-infrastructure` repository 🙂

Reply:
I added the relative samples and validation in 6db74b991261d24f9ee8f23f072340b482bd102a and opened [a PR on the infra repo](https://github.com/safe-global/safe-infrastructure/pull/150).